frontend/alert/components/GroupSuggestion.tsx

```diff
@@ -0,0 +1,357 @@
+     * Used to make sure to update button states when manually selecting each section
+     * @param activity - section activity
+     */
+    const selectedAllActivity = (activity) => {
+        // check if the user selected all courses of a certain activity in a group suggestion
+        let selectedAll = true;
+
+        sections[activity].map((section) => {
+            if (!selectedCourses.has(section)) {
+                selectedAll = false;
+            }
+        });
+
+        syncButtonStates(activity, selectedAll);
+        return selectedAll;
+    };
+
+    /**
+     * Update button states to match selected courses in the case the user is manually checking/unchecking sections
+     * Ex. allLectures should be true when user manually select all "LEC" sections
+     * @param activity - section activity
+     */
+    const syncButtonStates = (activity, correctState) => {
+        // update button state if its not the same as correctState
+        if (buttonStates[activity] != correctState) {
+            const newButtonStates = { ...buttonStates };
+            newButtonStates[activity] = correctState;
+            setButtonStates(newButtonStates);
+        }
+    };
+
+    /**
+     * Takes the activity of button clicked: LEC, REC, LABS, etc and switch state
+     * @param activity - section activity
+     */
+    const toggleButton = (activity) => {
+        const newButtonStates = { ...buttonStates };
+
+        newButtonStates[activity] = !newButtonStates[activity];
+        syncCheckAlls(activity, newButtonStates[activity], setSelectedCourses);
+
+        setButtonStates(newButtonStates);
+    };
+
+    /**
+     * If one of the select all is toggled, update selected courses to match
+     * @param activity - section activity
+     * @param checkedAll - whether all sections of activity should be checked or not
+     * @param setSelectedCourses - method to update selected courses set
+     */
+    const syncCheckAlls = (activity, checkedAll, setSelectedCourses) => {
+        const newSelectedCourses = new Set(selectedCourses);
+
+        if (sections[activity]) {
+            sections[activity].map((section) => {
+                // checkedAll = false but selectedCourses has course
+                if (newSelectedCourses.has(section) && !checkedAll) {
+                    newSelectedCourses.delete(section);
+
+                    // checkedAll = true but selectedCourses doesn't have course
+                } else if (!newSelectedCourses.has(section) && checkedAll) {
+                    newSelectedCourses.add(section);
+                }
+            });
+            setSelectedCourses(newSelectedCourses);
+        }
+    };
+
+    /**
+     * Update the selected courses set when user check/uncheck box
+     * @param selectedCourses - selected courses set
+     * @param setSelectedCourses - method to update selected courses set
+     * @param setSelectedCourses - the section
```

Comment:
I think this should be `suggestion` not `setSelectedCourses` (duplicated param)?